src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java

```diff
@@ -15,7 +16,9 @@
     @Override
     public User login(String login, String password) throws AuthenticationException {
         Optional<User> optionalUser = userService.findByLogin(login);
-        if (optionalUser.isPresent() && optionalUser.get().getPassword().equals(password)) {
+        byte[] salt = optionalUser.get().getSalt();
+        String saltedPassword = HashUtil.hashPassword(password, salt);
+        if (optionalUser.isPresent() && optionalUser.get().getPassword().equals(saltedPassword)) {
             return optionalUser.get();
         }
```

Comment:
on line 19 you are calling get(), and only on line 21 you are checking if Optional is present